include/cudaq/Optimizer/Transforms/Passes.td

```diff
@@ -471,4 +471,20 @@ def MultiControlDecompositionPass: Pass<"multicontrol-decomposition",
   }];
 }
 
+def MappingPass: Pass<"mapping", "mlir::func::FuncOp"> {
+  let summary = "Perform qubit mapping to account for connectivity constraints";
+  let description = [{
+    Some backends cannot support any-to-any multi-qubit operations, so this pass
+    performs mapping of the qubits by inserting the necessary swap operations
+    into the Quake IR.
+
+    Note: this pass requires value semantics. It will throw an error if the
+    Quake is not in value semantics form.
```

Comment:
I think this is actually meant in a strict sense. The IR can fall along a spectrum from being strictly memory reference semantics (no `wire` or `control` types) to strictly register value semantics (no `ref` or `veq` types).

Maybe re-word this a bit to make it clearer?

Reply:
Updated in 9f60585e